library/bignum_mod.h

```diff
@@ -173,7 +173,51 @@ void mbedtls_mpi_mod_modulus_free( mbedtls_mpi_mod_modulus *m );
 /* END MERGE SLOT 6 */
 
 /* BEGIN MERGE SLOT 7 */
+/** Read public representation data stored in a buffer into a residue structure.
```

Comment:
To my understanding we cannot guarantee that since the context passed in the data conversion is passed by moduli. So I will be ammeding the description to make it more clear.

Reply:
“public representation data” is determined by the modulus. Each option in `mbedtls_mpi_mod_ext_rep` defines the representation (for now it is just byte order, will add new types for Montgomery curve style masked keys later).

`mbedtls_mpi_mod_write` followed by `mbedtls_mpi_mod_read`, both of them called with the same modulus should yield the same residue, with the same size. Number of leading zeroes are determined by the modulus and the limb size.

We are planning to use these on any standard input data like certificate fields, internal key storage, fields in SSL/TLS messages, input passed to the PSA interface.